backend/src/models/cost_distribution.py

```diff
@@ -0,0 +1,36 @@
+from ..config.extensions import db
+
+
+class Cost_distribution(db.Model):
+    cost_distribution_id = db.Column(db.Integer, primary_key=True)
+    rental_id = db.Column(db.Integer,db.ForeignKey('rental.rental_id'),
+                           nullable=False )
+    vehicle_cost = db.Column(db.Float, nullable=False)
+    item_cost = db.Column(db.Float, nullable=True)
+    insurance_cost = db.Column(db.Float ,nullable=True)
+    penalty_charges = db.Column(db.Float ,nullable=True)
+    total = db.Column(db.Float ,nullable=False)
+
+    rental = db.relationship('Rental', backref='costs')
+
+    def get_id(self):
+        return self.cost_distribution_id    
+
+    def __init__(self, cost_dict):
+        self.rental_id = cost_dict["rental_id"]
+        self.vehicle_cost = cost_dict["vehicle_cost"]
+        self.item_cost = cost_dict["item_cost"]
+        self.insurance_cost = cost_dict["insurance_cost"]
+        self.penalty_charges = cost_dict["penalty_charges"]
+        self.total = self.calculate_total()
+
+    def calculate_total(self):
+        return change(self.vehicle_cost)+change(self.item_cost)+change(self.insurance_cost)+change(self.penalty_charges)
+    
+    def serialize(self):
+        return {"cost_distribution_id": self.cost_distribution_id, "rental_id": self.rental_id, "vehicle_cost": self.vehicle_cost,"item_cost": self.item_cost,"insurance_cost": self.insurance_cost, "penalty_charges": self.penalty_charges, "total": self.total}
+    
+def change(smthing):
```

Comment:
Co robi ta funkcja? Nie jestem w stanie zrozumieć, może jakaś inna nazwa dla parametru?

Reply:
w tej tabeli jest wiele wartosci nullable true, więc żeby móc potem policzyć sume to tak to zostało ograne, żeby wpisywać za pomocą tej funkcji "0" zamiast "null"

Reply:
No to w takim razie zmień  nazwy na bardziej intuicyjne

Reply:
done

Reply:
Średnio, ale niech już będzie